pro/src/pages/AdageIframe/app/components/OffersInstantSearch/OffersSearch/FiltersTags/FiltersTags.tsx

```diff
@@ -162,18 +161,15 @@ export const FiltersTags = ({
           handleSubmit()
         })
       )}
-      {
-        // TODO : We should use a custom isEqual instead of lodash one
```

Comment:
Pourquoi on ne veut pas utiliser isEqual ? Est-ce que tu as l'historique de ce TODO ? Je vois qu'on l'utilise une dizaine de fois ailleurs dans pc pro

Reply:
C'était l'objectif du Todo 🤔 qui me paraissait cohérent pour le coup puisque `isEqual` fait une vérification en profondeur et avancée des deux objets à comparer, alors qu'ici on souhaite simplement vérifier si les propriétés d'un objet (dont on connaît la structure) est vide.

Un test de performance effectué sur les deux approches permet de confirmer que l'utilisation d'une fonction manuelle dans ce cas est ~99% plus rapide qu'utilise `lodash.isEqual` (le test que je viens de rédiger en question : https://jsperf.app/likapa)

Pour les autres cas de figure dans le NPP, je n'ai pas été voir (car il n'y avait aucune annotation TODO) mais d'après ce que je vois, il y en a beaucoup où l'on gagnerait en perfs à faire une comparaison manuelle.

Reply:
On avait une fonction qui faisait un deep check pour vérifier qu'un objet soit strictement égal à un autre. Au vu du besoin qui est de simplement vérifier que des filtres soient vides, je trouve que c'est inutile et lourd car `isEqual` fait beaucoup de choses en interne.
Le test de perfs avait pour objectif de souligner cela.

En soit on a le choix : 
- soit on compare avec des valeurs vides sous forme de primitives, ce qui correspond au besoin actuel et allège au maximum l'exécution.
- soit on rédige une nouvelle fonction qui fait un deep check précis des propriétés de `values` pour prévoir le cas de figure où on déciderait plus tard d'avoir des valeurs par défaut dans les arrays.

Reply:
> Au vu du besoin qui est de simplement vérifier que des filtres soient vides

C'est là où je ne suis pas forcément d'accord. On veut comparer avec l'objet de la requête initiale, qui là, est vide.

Reply:
Par "objet de la requête initiale", tu entends la constante `ADAGE_FILTERS_DEFAULT_VALUES` ?

Reply:
Oui, c'est ça. On veut comparer à une constante qui peut changer.

Reply:
> Oui, c'est ça. On veut comparer à une constante qui peut changer.

… dont les **valeurs de propriété** peuvent changer, et non la structure.

Sauf si plus tard de nouveaux filtres font leur apparition, il faudra effectivement aller modifier cette fonction, mais de mon point de vue c'est préférable car c'est mieux maîtriser le besoin et éviter des fonctions trop lourdes à chaque re-render.

Si `lodash.isEqual` était "juste" suffisant je comprendrais qu'on le laisse, mais en l'occurence c'est un bazooka cette fonction. Je pense que c'était l'idée du TODO initial : "_use a custom isEqual instead of lodash one_"

La modification que j'avais proposée [plus haut](https://github.com/pass-culture/pass-culture-main/pull/14162#discussion_r1762998541) n'est certes plus étroitement liée à la constante `ADAGE_FILTERS_DEFAULT_VALUES` mais elle reste lisible, facilement modifiable si un jour les filtres par défaut changent, et peu coûteuse en perfs sur de nombreux re-renders.
Je proposerai même pour rester cohérent et ne pas faire de mix carrément vérifier avec des les primitives que l'on veut :

```tsx
const areFiltersEquals = (filters: SearchFormValues) => {
  return (
    // Primitives defaults
    filters.eventAddressType === OfferAddressType.OTHER &&
    filters.geolocRadius === 50 &&

    // Array defaults (empty)
    filters.domains.length === 0 &&
    filters.departments.length === 0 &&
    filters.academies.length === 0 &&
    filters.formats.length === 0 &&
    filters.students.length === 0 &&

    // venue default is null
    filters.venue === null
  )
}
```